usermods/Bar_Timer/usermod_timer_bar.h

```diff
@@ -0,0 +1,210 @@
+#pragma once
+
+#include "wled.h"
+
+#define RUNNING 1
+#define STOPPED 0
+
+/**
+ * This class create a timer using the connected strip. When the timer starts, the strip is filled with the color of the solid effect.
+ * Led are turned off according to the remaining time of the timer.
+ * MQTT messages to use the timer are:
+ * 
+ * 1) (device topic)/timerbar with payload "+X" -> X is the duration of the timer in seconds. If there's no active timer, with this command 
+ * the timer starts with a duration of X. If there is already a running timer, the same command adds X seconds to timer. In this case the strip is
+ * not filled again, but leds will start to turn off slower and the already turned off leds will remain off
+ * 
+ * 2) (device topic)/timerbar with payload "STOP" -> This message ends the timer and the strip.
+ * 
+ * The strip will answer with this messages:
+ * 
+ * 1) (device topic)/timerbar/status with payload "START" -> Feedback to know the timer actually started
+ * 2) (device topic)/timerbar/status with payload "END" -> To let know the timer has ended
+ * 
+ * Everytime the timer ends, the strip returns in the state before the timer start command (effect, palette...)
+*/
+
+
+
+
+class UsermodTimerBar : public Usermod {
+
+    private:
+
+        uint8_t timer_status;
+        uint16_t number_of_leds, led_turn_off = 0;
+        uint32_t millis_wait_time; 
+        uint16_t start_time, end_time;
+        Segment main;
+        byte settings[3];
+        int helper  = 0;
+
+
+        void stop_timer() {
+
+          led_turn_off = 0;
+          
+          //Restore previous strip settings
+          effectSpeed = settings[1];
+          effectPalette = settings[2];
+          strip.setMode(0, effectCurrent);
+
+          publishMqtt("END", false);
+
+          timer_status = STOPPED;
+        }
+
+
+        void check_timer() {
+            
+          end_time = millis();
+
+          if ((end_time - start_time) >= millis_wait_time) {
+
+            led_turn_off++;
+            start_time = end_time;
+
+            if (led_turn_off == number_of_leds) {
+
+              stop_timer();
+            }
+          }         
+        }        
+
+
+
+    public:
+
+        void setup() {
+
+          timer_status = STOPPED;
+          led_turn_off = 0;
+        }
+    
+        void loop (){
+
+          if (timer_status == RUNNING) {
+
+            helper = 1;
+            strip.trigger();
+            strip.show();
```

Comment:
Never call `strip.show()` from usermod. It can mess other things.
`strip.trigger()` should be enough to force LED update on next cycle. And IMO even that is unnecessary if other best practices are followed.